### ISSUE TITLE
[bug] Attack the characters in the vehicle directly over the armor

### ISSUE THREAD

**Issue by xingchengs** [DonBruce64/MinecraftTransportSimulator#1223]
Scape and run: mod creatures such as parasites can directly cross the armor of the vehicle and attack the characters in the vehicle

**Comment by xingchengs** [comment]
Test:  Scape and Run: Parasites

**Comment by DonBruce64** [comment]
Do they physically move though the vehicle, or do they just attack you through it?

**Comment by xingchengs** [comment]
If you attack me outside the car, you can directly attack the characters inside the car



---Original---
From: ***@***.***&gt;
Date: Sun, Jun 12, 2022 14:55 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [DonBruce64/MinecraftTransportSimulator] [bug] Attack thecharacters in the vehicle directly over the armor (Issue #1223)




 
Do they physically move though the vehicle, or do they just attack you through it?

**Comment by DonBruce64** [comment]
That doesn't make sense.  Are you inside or outside the car?

**Comment by xingchengs** [comment]
In car



---Original---
From: ***@***.***&gt;
Date: Mon, Jun 13, 2022 04:17 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [DonBruce64/MinecraftTransportSimulator] [bug] Attack thecharacters in the vehicle directly over the armor (Issue #1223)




 
That doesn't make sense.  Are you inside or outside the car?

**Comment by DonBruce64** [comment]
But you said "If you attack me outside the car".  That means you are outside the car and being attacked?  Or are you inside the car, and the creature is outside the car, and they can attack you?  If so, which car?  Cause a lot of them don't have boxes to prevent attacks.

**Comment by xingchengs** [comment]
I'm in the car, and the enemy creature is outside the car. The enemy creature can directly attack me over the armor of the vehicle。
Mod:immersive vehicles/iv/mts] veb automobilwerke schwikau military/civil



---Original---
From: ***@***.***&gt;
Date: Mon, Jun 13, 2022 12:20 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [DonBruce64/MinecraftTransportSimulator] [bug] Attack the characters in the vehicle directly over the armor (Issue #1223)




 
But you said "If you attack me outside the car".  That means you are outside the car and being attacked?  Or are you inside the car, and the creature is outside the car, and they can attack you?  If so, which car?  Cause a lot of them don't have boxes to prevent attacks.

**Comment by DonBruce64** [comment]
Which vehicle?

**Comment by xingchengs** [comment]
All vehicles have this problem, not an exception



---Original---
From: ***@***.***&gt;
Date: Mon, Jun 13, 2022 20:47 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [DonBruce64/MinecraftTransportSimulator] [bug] Attack the characters in the vehicle directly over the armor (Issue #1223)




 
Which vehicle?

**Comment by DonBruce64** [comment]
Then this isn't a bug.  Most vehicles don't have hitboxes to block mobs from attacking the player.  Only a very few do, like tanks and AA vehicles.

**Comment by DonBruce64** [comment]
After testing with the mod, this is the conclusion:
depiction of what is happening, red are hitboxes
![image](https://user-images.githubusercontent.com/15241611/177052762-da57f736-92ed-4d63-9903-3147e0654130.png)

this is what was happening in the report
since to the ai, theres no obstacles in the way, so it has direct sight with the player
![image](https://user-images.githubusercontent.com/15241611/177052797-aa74037d-9939-4c1b-acd5-8807e634c72c.png)


In conclusion, hitboxes are working correctly for armor.  It's just that not all vehicles have said armor.
